x-pack/plugins/apm/public/components/shared/FilterBar/DatePicker.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import datemath from '@elastic/datemath';
+import {
+  CommonProps,
+  EuiSuperDatePicker,
+  EuiSuperDatePickerProps
+} from '@elastic/eui';
+import { memoize } from 'lodash';
+import React from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { updateTimePicker } from 'x-pack/plugins/apm/public/store/urlParams';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+
+interface Props extends RouteComponentProps {
+  dispatchUpdateTimePicker: typeof updateTimePicker;
+}
+
+interface DatePickerParams {
+  rangeFrom: string;
+  rangeTo: string;
+  refreshPaused: boolean;
+  refreshInterval: number;
+}
+
+// TODO this can be removed when EUI PR lands: https://github.com/elastic/eui/pull/1574
+declare module '@elastic/eui' {
+  interface OnTimeChangeProps {
+    start: string;
+    end: string;
+    isInvalid: boolean;
+    isQuickSelection: boolean;
+  }
+
+  interface OnRefreshChangeProps {
+    isPaused: boolean;
+    refreshInterval: number;
+  }
+
+  export type EuiSuperDatePickerProps = CommonProps & {
+    start?: string;
+    end?: string;
+    isPaused?: boolean;
+    refreshInterval?: number;
+    onTimeChange: (props: OnTimeChangeProps) => void;
+    onRefreshChange?: (props: OnRefreshChangeProps) => void;
+    showUpdateButton?: boolean;
+  };
+  export const EuiSuperDatePicker: React.SFC<EuiSuperDatePickerProps>;
+}
+
+const APM_DEFAULT_TIME_OPTIONS: DatePickerParams = {
+  rangeFrom: 'now-24h',
+  rangeTo: 'now',
+  refreshPaused: true,
+  refreshInterval: 0
+};
+
+class DatePickerComponent extends React.Component<Props> {
+  public refreshTimeoutId = 0;
+
+  public getParamsFromSearch = memoize((search: string) => {
+    const query = toQuery(search);
+    const converted = {
+      ...query,
+      refreshPaused:
+        query.refreshPaused === 'true' || query.refreshPaused === true,
+      refreshInterval:
+        typeof query.refreshInterval === 'undefined'
+          ? undefined
+          : Number(query.refreshInterval)
```

Comment:
We already have a `toNumber` utility method. Can you use that?
https://github.com/jasonrhodes/kibana/blob/e6f30167d67f030c180b098d5f3a4d20fa15622b/x-pack/plugins/apm/public/store/urlParams.ts#L87-L92

Reply:
Sure. Why does that helper accept an array of strings and convert it to undefined?

Reply:
> Why does that helper accept an array of strings and convert it to undefined?

It's a very good question. I git blamed and found myself guilty. Unfortunately I can't remember why and I didn't add a test :( Looking at the usages of `toNumber` I think it's safe to remove the array handling, and simply do

```
export function toNumber(value?: string) {
  if (value !== undefined) {
    return parseInt(value, 10);
  }
}
```

Reply:
Hmm, it's actually something you did in https://github.com/elastic/kibana/pull/24294/files.
And `toString` was also changed to handle arrays. I think neither should handle arrays tbh.

Reply:
And while we are added:
```
kuery: legacyDecodeURIComponent(kuery as string | undefined),
```

can be changed back to:
```
kuery: legacyDecodeURIComponent(kuery),
```